Adding cap-helm-charts-ci pipeline.

bikramnehra opened the pull request:
This pipeline is responsible for opening a PR against SUSE helm charts repo for the latest kubecf and cf-operator helm chart releases.

Commit:
adding cap-helm-charts-ci pipeline.

This pipeline is responsible for opening a PR against SUSE helm charts
repo for the latest kubecf and cf-operator releases.

Commit:
removing private key and some formatting.

Commit:
updating file path.

Commit:
updating README

viovanov commented on cap-helm-charts-ci/vars.yaml.template:
```diff
@@ -0,0 +1,14 @@
+---
+ci-uri: https://github.com/SUSE/cap.git
+ci-branch: master
+bucket: <CAP-RELEASE-BUCKET> # S3 bucket to pick the release from.
```
should contain the actual bucket name

bikramnehra replied:
DONE.

viovanov commented on cap-helm-charts-ci/vars.yaml.template:
```diff
@@ -0,0 +1,14 @@
+ci-uri: https://github.com/SUSE/cap.git
+ci-branch: master
+bucket: <CAP-RELEASE-BUCKET> # S3 bucket to pick the release from.
+region: <CAP-RELEASE-BUCKET-REGION> # Region of the bucket.
```
actual value please

bikramnehra replied:
DONE.

viovanov commented on cap-helm-charts-ci/vars.yaml.template:
```diff
@@ -0,0 +1,14 @@
+region: <CAP-RELEASE-BUCKET-REGION> # Region of the bucket.
+access-key-id: <S3-ACCESS-KEY-ID> # Access key for the release bucket.
+secret-access-key: <S3-SECRET-ACCESS-KEY> # Secret Key for the release bucket.
+helm-charts-repo-uri: <HELM-CHART-REPO-URI> # URI for helm chart repo.
```
actual value please

bikramnehra replied:
DONE.

viovanov commented on cap-helm-charts-ci/vars.yaml.template:
```diff
@@ -0,0 +1,14 @@
+access-key-id: <S3-ACCESS-KEY-ID> # Access key for the release bucket.
+secret-access-key: <S3-SECRET-ACCESS-KEY> # Secret Key for the release bucket.
+helm-charts-repo-uri: <HELM-CHART-REPO-URI> # URI for helm chart repo.
+helm-charts-repo-branch: <HELM-CHART-REPO-BRANCH> # Branch for helm chart repo.
```
actual value please

bikramnehra replied:
DONE.

viovanov commented on cap-helm-charts-ci/vars.yaml.template:
```diff
@@ -0,0 +1,14 @@
+helm-charts-repo-uri: <HELM-CHART-REPO-URI> # URI for helm chart repo.
+helm-charts-repo-branch: <HELM-CHART-REPO-BRANCH> # Branch for helm chart repo.
+helm-charts-repo-private-key: <HELM-CHARTS-REPO-PRIVATE-KEY> # Github private key for helm chart repo.
+github-token: <GITHUB-TOKEN> # Your GitHub token.
```
For secrets, you could assume the user has the blackbox in `BLACKBOX_DIR` and get the values in a script.
Otherwise this is error prone, and might leak secrets all over the place, depending on the user's carefulness.

bikramnehra replied:
DONE.

viccuad replied:
Another option is adding the secret itself into the concourse.suse.dev instance (see [here](https://github.com/SUSE/cloudfoundry/tree/master/concourse.suse.dev#manipulating-concourse-pipeline-secrets)) and consuming it ala `((secret))`.

This has the benefit that the pipeline yaml doesn't contain a plain secret (which could be seen by `fly get-pipeline`), nor fly hijacking allows you to see the secret. But on the pipeline build, the value will be correctly used.

viovanov commented on cap-helm-charts-ci/tasks/create_pr.sh:
```diff
@@ -0,0 +1,67 @@
+
+# Setup git
+mkdir -p ~/.ssh
+echo "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=="  ...
```
This is not safe - is `ssh-keyscan` an option?

bikramnehra replied:
DONE.

Commit:
fixing travis failures

Commit:
Implementing review comments.

Update the vars.yaml.template with real values.

Commit:
Implementing review comments.

1. Switched to gomplate.
2. Secret Management directyly through blackbox.
3. updated README.
